pinot-spi/src/main/java/org/apache/pinot/spi/env/PinotConfiguration.java

```diff
@@ -170,8 +173,15 @@ public static List<Configuration> applyDynamicEnvConfig(List<Configuration> conf
       Map<String, String> environmentVariables) {
     return configurations.stream().peek(configuration -> {
       for (String dynamicEnvConfigVarName : configuration.getStringArray(ENV_DYNAMIC_CONFIG_KEY)) {
-        configuration.setProperty(dynamicEnvConfigVarName,
-            environmentVariables.get(configuration.getString(dynamicEnvConfigVarName)));
+        //if the environment variable doesn't exist or the property is already checked do not add the property
```

Comment:
Seems we are not checking whether the property already exists. Is this expected? If so, let's modify the comments

Reply:
Changed the description